File: grader_examples/grader.pas
uses NomeProblema;

var	fr, fw : text;

{ Declaring variables }
	N : Integer;
	M : Integer;
	S : Integer;
	P : array of Integer;
	from : array of Integer;
	too : array of Integer;
	length : array of Integer;
	H : Integer;
	W : Integer;
	R : array of array of Char;
	G : array of array of Char;
	B : array of array of Char;
	res : Integer;
	scelti : array of Integer;
	colore : array of Double;

{ iterators used in for loops }
	i0, i1: Integer;
	
begin
{$ifdef EVAL}
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
{$else}
    fr := input;
    fw := output;
{$endif}
    reset(fr);
    rewrite(fw);

	{ Reading input }
	readln(fr, N, M, S);
	Setlength(P, N);
	for i0 := 0 to N do
	begin
		read(fr, P[i0]);
	end;
	Setlength(from, M);
	Setlength(too, M);
	Setlength(length, M);
	for i0 := 0 to M do
	begin
		read(fr, from[i0], too[i0], length[i0]);
	end;
	readln(fr, H, W);
	Setlength(R, H, W);
	Setlength(G, H, W);
	Setlength(B, H, W);
	for i0 := 0 to H do
	begin
		for i1 := 0 to W do
		begin
			read(fr, R[i0][i1], G[i0][i1], B[i0][i1]);
		end;
	end;

	{ Calling functions }
	contapersone(M, from, too);
	sceglicolori(res, scelti, colore);

	{ Writing output }
	writeln(fw, res);
	for i0 := 0 to res do
	begin
		writeln(fw, scelti[i0], colore[i0]);
	end;
	writeln(fw);
	
	close(fr);
    close(fw);
end.

File: grader_examples/grader.c
#include <stdio.h>
#include <assert.h>
#include <stdlib.h>

static FILE *fr, *fw;

// Declaring variables
static int N;
static int M;
static int S;
static int* P;
static int* from;
static int* too;
static int* length;
static int H;
static int W;
static char** R;
static char** G;
static char** B;
static int res;
static int* scelti;
static double* colore;

// Declaring functions
int contapersone(int M, int* from, int* too);
void sceglicolori(int res, int* scelti, double* colore);

int main() {
	#ifdef EVAL
		fr = fopen("input.txt", "r");
		fw = fopen("output.txt", "w");
	#else
		fr = stdin;
		fw = stdout;
	#endif

	// Iterators used in for loops
	int i0, i1;

	// Reading input
	fscanf(fr, "%d %d %d ", &N, &M, &S);
	P = (int*)malloc(N * sizeof(int));
	for (i0 = 0; i0 < N; i0++) {
		fscanf(fr, "%d ", &P[i0]);
	}
	from = (int*)malloc(M * sizeof(int));
	too = (int*)malloc(M * sizeof(int));
	length = (int*)malloc(M * sizeof(int));
	for (i0 = 0; i0 < M; i0++) {
		fscanf(fr, "%d %d %d ", &from[i0], &too[i0], &length[i0]);
	}
	fscanf(fr, "%d %d ", &H, &W);
	R = (char**)malloc(H * sizeof(char*));
	for (i0 = 0; i0 < H; i0++) {
		R[i0] = (char*)malloc(W * sizeof(char));
	}
	G = (char**)malloc(H * sizeof(char*));
	for (i0 = 0; i0 < H; i0++) {
		G[i0] = (char*)malloc(W * sizeof(char));
	}
	B = (char**)malloc(H * sizeof(char*));
	for (i0 = 0; i0 < H; i0++) {
		B[i0] = (char*)malloc(W * sizeof(char));
	}
	for (i0 = 0; i0 < H; i0++) {
		for (i1 = 0; i1 < W; i1++) {
			fscanf(fr, "%c %c %c ", &R[i0][i1], &G[i0][i1], &B[i0][i1]);
		}
	}

	// Calling functions
	res = contapersone(M, from, too);
	scelti = (int*)malloc(res * sizeof(int));
	colore = (double*)malloc(res * sizeof(double));
	sceglicolori(res, scelti, colore);

	// Writing output
	fprintf(fw, "%d\n", res);
	for (i0 = 0; i0 < res; i0++) {
		fprintf(fw, "%d %lf\n", scelti[i0], colore[i0]);
	}
	for (i0 = 0; i0 < H; i0++) {
		for (i1 = 0; i1 < W; i1++) {
			fprintf(fw, "%c", R[i0][i1]);
		}
		fprintf(fw, "\n");
	}
	
	fclose(fr);
	fclose(fw);
	return 0;
}
